docs/apriori_sbms.ipynb

```diff
@@ -0,0 +1,384 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# load some elegans witvilet data\n",
+    "base = '../json_connectomes/witvilet/'\n",
+    "graph_files = sorted(os.listdir(base))\n",
+    "wit_connectomes = []\n",
+    "for f in graph_files:\n",
+    "    # These contain multiple synapse types, and thus can have multiedges and need to be flattened.\n",
+    "    # Alternatively, we could preform the analysis separately for each synapse type\n",
+    "    wit_connectomes.append(GraphIO.flatten_multigraph(GraphIO.load(os.path.join(base, f))[0]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Now we will fit sbms for each grouping for each connectome."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "groups = [[\"hemisphere\"], [\"cell_type0\"], [\"hemisphere\", \"cell_type0\"]]\n",
+    "rows = {}\n",
+    "for i in range(len(wit_connectomes)):\n",
+    "    g_row = {}\n",
+    "    for group in groups:\n",
+    "        g_row[str(group)] = fit_apriori_sbm(wit_connectomes[i], group, plot_sbms=False)\n",
+    "    rows['witvilet_' + str(i)] = g_row\n",
+    "wit_results = pd.DataFrame.from_dict({(i,j): rows[i][j]\n",
+    "                           for i in rows.keys()\n",
+    "                           for j in rows[i].keys()},\n",
+    "                       orient='index')\n",
+    "wit_results.head(24)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's plot the BIC and Likelihood"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots(8, 2)\n",
+    "out_index = wit_results.index.levels[0]\n",
+    "colors = ['red', 'blue', 'green', 'orange', 'purple', 'grey', 'cyan', 'pink']\n",
+    "for i, key in enumerate(out_index):\n",
+    "    data = wit_results.loc[key]\n",
+    "    ax[i, 0].scatter(data['bic'].index, data['bic'], label=key, color=colors[i])\n",
+    "    ax[i, 1].scatter(data['liklihood'].index, data['liklihood'], label=key, color=colors[i])\n",
+    "    ax[i, 0].set(xlabel='Groups, stage ' + str(i), ylabel='-BIC')\n",
+    "    ax[i, 1].set(xlabel='Groups, stage ' + str(i), ylabel='Likelihood')\n",
+    "plt.ylabel('-BIC \\t Likelihood')\n",
+    "fig.suptitle('BIC and Likelihood: C. Elegans at Different Developmental Stages')\n",
+    "fig.set_size_inches(16, 32)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once again, we find that cell type explains the topology the best. This is still not entirely surprising, since the\n",
```

Comment:
This is a lot of figures, each with very few datapoints.  Maybe you can combine them somehow?  Perhaps use a color map with different levels of brightness based on age (like I did in GraphStats)?

Reply:
Originally they where combined, but the issue is BIC is on very different scales for each developmental stage. Pedigo thought it wasn't really reasonable to compare across the different  connectomes

Reply:
What if you normalized each dev stage so that the lowest BIC is -1?  I get why you can't really compare across dev stages, but it seems reasonable to compare rankings right?

Reply:
Ok I normalized to 0, they look pretty good